framework/core/Project/Lang.pm

```diff
@@ -56,15 +57,34 @@ sub new {
     return $class->SUPER::new($PID, $name, $vcs);
 }
 
+sub printstack {
+    my ($package, $filename, $line, $subroutine, $hasargs, $wantarray, $evaltext, $is_require, $hints, $bitmask, $hinthash);
+    my $i = 1;
+    my @r;
+    while (@r = caller($i)) {
+        ($package, $filename, $line, $subroutine, $hasargs, $wantarray, $evaltext, $is_require, $hints, $bitmask, $hinthash) = @r;
+        print "$filename:$line $subroutine\n";
+        $i++;
+    }
+}
+
 #
 # Determines the directory layout for sources and tests
 #
 sub determine_layout {
     @_ == 2 or die $ARG_ERROR;
-    my ($self, $rev_id) = @_;
-    my $dir = $self->{prog_root};
-    my $result = _layout1($dir) // _layout2($dir);
-    die "Unknown layout for revision: ${rev_id}" unless defined $result;
+    my ($self, $revision_id) = @_;
+    my $work_dir = $self->{prog_root};
+
+    # Only two sets of layouts in this case
+    my $result;
+    if (-e "$work_dir/src/main"){
+      $result = {src=>"src/main/java", test=>"src/test/java"};
+    }
+    if (-e "$work_dir/src/java"){
+      $result = {src=>"src/java", test=>"src/test"};
+    }
+    die "Unknown layout for revision: ${revision_id}" unless defined $result;
```

Comment:
Is this change necessary?

Reply:
With the reorg of the Lang project _layout2() was no longer correct.  Rather than fix it I changed determine_layout to be identical to 5 other projects with the same layout choices.